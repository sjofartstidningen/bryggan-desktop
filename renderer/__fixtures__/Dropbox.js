/**
 * Fixture response from /files/list_folder
 *
 * Update fixture with the following command:
 *  curl -X POST https://api.dropboxapi.com/2/files/list_folder \
 *    --header "Authorization: Bearer <api_key>" \
 *    --header "Content-Type: application/json" \
 *    --data "{\"path\": \"/Tidningen/2018/11\",\"recursive\": false}" > res.json
 */
const filesListFolder = {
  entries: [
    {
      '.tag': 'folder',
      name: 'A',
      path_lower: '/tidningen/2018/11/a',
      path_display: '/Tidningen/2018/11/A',
      parent_shared_folder_id: '1196849465',
      id: 'id:8cfsOlDXQ0AAAAAAAABfZA',
      sharing_info: {
        read_only: false,
        parent_shared_folder_id: '1196849465',
        traverse_only: false,
        no_access: false,
      },
    },
    {
      '.tag': 'folder',
      name: 'D',
      path_lower: '/tidningen/2018/11/d',
      path_display: '/Tidningen/2018/11/D',
      parent_shared_folder_id: '1196849465',
      id: 'id:8cfsOlDXQ0AAAAAAAABfew',
      sharing_info: {
        read_only: false,
        parent_shared_folder_id: '1196849465',
        traverse_only: false,
        no_access: false,
      },
    },
    {
      '.tag': 'folder',
      name: 'B',
      path_lower: '/tidningen/2018/11/b',
      path_display: '/Tidningen/2018/11/B',
      parent_shared_folder_id: '1196849465',
      id: 'id:8cfsOlDXQ0AAAAAAAABf0w',
      sharing_info: {
        read_only: false,
        parent_shared_folder_id: '1196849465',
        traverse_only: false,
        no_access: false,
      },
    },
    {
      '.tag': 'folder',
      name: 'C',
      path_lower: '/tidningen/2018/11/c',
      path_display: '/Tidningen/2018/11/C',
      parent_shared_folder_id: '1196849465',
      id: 'id:8cfsOlDXQ0AAAAAAAABf1A',
      sharing_info: {
        read_only: false,
        parent_shared_folder_id: '1196849465',
        traverse_only: false,
        no_access: false,
      },
    },
    {
      '.tag': 'folder',
      name: 'Framvagn',
      path_lower: '/tidningen/2018/11/framvagn',
      path_display: '/Tidningen/2018/11/Framvagn',
      parent_shared_folder_id: '1196849465',
      id: 'id:8cfsOlDXQ0AAAAAAAABgdw',
      sharing_info: {
        read_only: false,
        parent_shared_folder_id: '1196849465',
        traverse_only: false,
        no_access: false,
      },
    },
    {
      '.tag': 'folder',
      name: 'till tryckeriet',
      path_lower: '/tidningen/2018/11/till tryckeriet',
      path_display: '/Tidningen/2018/11/till tryckeriet',
      parent_shared_folder_id: '1196849465',
      id: 'id:8cfsOlDXQ0AAAAAAAABhzA',
      sharing_info: {
        read_only: false,
        parent_shared_folder_id: '1196849465',
        traverse_only: false,
        no_access: false,
      },
    },
    {
      '.tag': 'file',
      name: 'ST_18_11.indd',
      path_lower: '/tidningen/2018/11/st_18_11.indd',
      path_display: '/Tidningen/2018/11/ST_18_11.indd',
      parent_shared_folder_id: '1196849465',
      id: 'id:8cfsOlDXQ0AAAAAAAABf2g',
      client_modified: '2018-11-27T15:14:14Z',
      server_modified: '2018-11-27T15:14:19Z',
      rev: '242cc47567939',
      size: 18907136,
      sharing_info: {
        read_only: false,
        parent_shared_folder_id: '1196849465',
        modified_by: 'dbid:AAAD--pzSalcq73PGbc9Ch9fgh0YPSvL1vQ',
      },
      content_hash:
        'd0ba27926e4dbdcbadfbd8658510bb33079eeff70d93f09bd837cd0d96fec31d',
    },
    {
      '.tag': 'file',
      name: 'Sid 23.pdf',
      path_lower: '/tidningen/2018/11/sid 23.pdf',
      path_display: '/Tidningen/2018/11/Sid 23.pdf',
      parent_shared_folder_id: '1196849465',
      id: 'id:8cfsOlDXQ0AAAAAAAABgmA',
      client_modified: '2018-11-29T11:54:26Z',
      server_modified: '2018-11-29T11:54:32Z',
      rev: '2437a47567939',
      size: 3526593,
      sharing_info: {
        read_only: false,
        parent_shared_folder_id: '1196849465',
        modified_by: 'dbid:AADI2C0iKz8PvrovDlpZ6DT3KypYuBReQss',
      },
      content_hash:
        '60a63157e14fbb182feca04fc129188d1348bddef8106c93efabfc6b4d3bde56',
    },
    {
      '.tag': 'file',
      name: 'ST_11_18_01_Framvagn.indd',
      path_lower: '/tidningen/2018/11/st_11_18_01_framvagn.indd',
      path_display: '/Tidningen/2018/11/ST_11_18_01_Framvagn.indd',
      parent_shared_folder_id: '1196849465',
      id: 'id:8cfsOlDXQ0AAAAAAAABgUg',
      client_modified: '2018-12-05T07:53:43Z',
      server_modified: '2018-12-05T07:53:48Z',
      rev: '246b647567939',
      size: 16314368,
      sharing_info: {
        read_only: false,
        parent_shared_folder_id: '1196849465',
        modified_by: 'dbid:AAAD--pzSalcq73PGbc9Ch9fgh0YPSvL1vQ',
      },
      content_hash:
        '7aa1cae47209ad8e2ff4aa2b34ecc8437e668b58adb88d21e88c7eb794abea12',
    },
    {
      '.tag': 'file',
      name: 'ST_11_18_05_D.indd',
      path_lower: '/tidningen/2018/11/st_11_18_05_d.indd',
      path_display: '/Tidningen/2018/11/ST_11_18_05_D.indd',
      parent_shared_folder_id: '1196849465',
      id: 'id:8cfsOlDXQ0AAAAAAAABgWA',
      client_modified: '2018-12-05T08:23:00Z',
      server_modified: '2018-12-05T08:23:09Z',
      rev: '246c447567939',
      size: 23642112,
      sharing_info: {
        read_only: false,
        parent_shared_folder_id: '1196849465',
        modified_by: 'dbid:AAAGsBZeMJWLDSD-bUhgCmyfDrXDDBlV-70',
      },
      content_hash:
        '50c5ab7c48a63d2f257c17e945b29334ef4bd5c42d064d9993fa2876447d5140',
    },
    {
      '.tag': 'file',
      name: 'ST_11_18_03_B.indd',
      path_lower: '/tidningen/2018/11/st_11_18_03_b.indd',
      path_display: '/Tidningen/2018/11/ST_11_18_03_B.indd',
      parent_shared_folder_id: '1196849465',
      id: 'id:8cfsOlDXQ0AAAAAAAABgVg',
      client_modified: '2018-12-05T09:03:13Z',
      server_modified: '2018-12-05T09:03:18Z',
      rev: '246d547567939',
      size: 18354176,
      sharing_info: {
        read_only: false,
        parent_shared_folder_id: '1196849465',
        modified_by: 'dbid:AAAD--pzSalcq73PGbc9Ch9fgh0YPSvL1vQ',
      },
      content_hash:
        'cde1429b2b1e8d43d6578bdf8f225252b3f7687892952ae9f1bc7084cd210cb6',
    },
    {
      '.tag': 'file',
      name: 'ST_11_18_04_C.indd',
      path_lower: '/tidningen/2018/11/st_11_18_04_c.indd',
      path_display: '/Tidningen/2018/11/ST_11_18_04_C.indd',
      parent_shared_folder_id: '1196849465',
      id: 'id:8cfsOlDXQ0AAAAAAAABgVw',
      client_modified: '2018-12-05T09:13:30Z',
      server_modified: '2018-12-05T09:13:35Z',
      rev: '2479847567939',
      size: 16797696,
      sharing_info: {
        read_only: false,
        parent_shared_folder_id: '1196849465',
        modified_by: 'dbid:AACCoALvX2HMOAqzduK1KNK1IPOb-MKGieI',
      },
      content_hash:
        '949e5c407cd7937ef407e25b25a4805ef67b58ba753c2a5d42fa844ae01e033c',
    },
    {
      '.tag': 'file',
      name: 'ST_11_18_02_A.indd',
      path_lower: '/tidningen/2018/11/st_11_18_02_a.indd',
      path_display: '/Tidningen/2018/11/ST_11_18_02_A.indd',
      parent_shared_folder_id: '1196849465',
      id: 'id:8cfsOlDXQ0AAAAAAAABgVA',
      client_modified: '2018-12-05T11:16:46Z',
      server_modified: '2018-12-05T11:16:52Z',
      rev: '2479c47567939',
      size: 19832832,
      sharing_info: {
        read_only: false,
        parent_shared_folder_id: '1196849465',
        modified_by: 'dbid:AAAD--pzSalcq73PGbc9Ch9fgh0YPSvL1vQ',
      },
      content_hash:
        '68aa69a67cda075d06025adea11827006623748ad1474be18eed1ed3707c68d6',
    },
    {
      '.tag': 'file',
      name: '~st_11_18_01_framva~y0mey8.idlk',
      path_lower: '/tidningen/2018/11/~st_11_18_01_framva~y0mey8.idlk',
      path_display: '/Tidningen/2018/11/~st_11_18_01_framva~y0mey8.idlk',
      parent_shared_folder_id: '1196849465',
      id: 'id:8cfsOlDXQ0AAAAAAAABiTA',
      client_modified: '2018-12-07T07:16:08Z',
      server_modified: '2018-12-07T07:16:14Z',
      rev: '247a847567939',
      size: 0,
      sharing_info: {
        read_only: false,
        parent_shared_folder_id: '1196849465',
        modified_by: 'dbid:AACrh22Y3qGiQTCRwwE3IHv4GvbjqNoznG8',
      },
      content_hash:
        'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855',
    },
  ],
  cursor:
    'AAGW04r7iXCgl1M02apsYHibNpqWwElo_Q-szs6wjYEGabyiMIjmnSxTSruDLWCa_AfNz0IM1Y06bjEyIraLVlymtGd7Nxtu3Oj3glSJR5TyHuaDEuPC6blwjv7_KsO8TljFZy1JclAuRClMiX5BA9uj2FTXYtAlN_pV6gB8rEk4ISYZuYt4gpTkQTMv7YLee9VTvBiLkDsngkAWeNni488exIaTq8xX3xSKQ_rvcc6PRnm7nQrc7HBeB3425qTQCAY',
  has_more: false,
};

const getAccount = {
  account_id: 'dbid:AACrh22Y3qGiQTCRwwE3IHv4GvbjqNoznG8',
  name: {
    given_name: 'Adam',
    surname: 'Bergman',
    familiar_name: 'Adam',
    display_name: 'Adam Bergman',
    abbreviated_name: 'AB',
  },
  email: 'adam.bergman@sjofartstidningen.se',
  email_verified: true,
  profile_photo_url:
    'https://dl-web.dropbox.com/account_photo/get/dbaphid%3AAAB-Yblu_z6W0CyArB8E-ymgpphnLEabMkk?size=128x128&vers=1460967974696',
  disabled: false,
  is_teammate: true,
  team_member_id: 'dbmid:AADhwrnyWESM_u0pImgwyWiB3lHn3ac4ioY',
};

const getCurrentAccount = {
  account_id: 'dbid:AACrh22Y3qGiQTCRwwE3IHv4GvbjqNoznG8',
  name: {
    given_name: 'Adam',
    surname: 'Bergman',
    familiar_name: 'Adam',
    display_name: 'Adam Bergman (Sj\u00f6fartstidningen)',
    abbreviated_name: 'AB',
  },
  email: 'adam.bergman@sjofartstidningen.se',
  email_verified: true,
  profile_photo_url:
    'https://dl-web.dropbox.com/account_photo/get/dbaphid%3AAAB-Yblu_z6W0CyArB8E-ymgpphnLEabMkk?size=128x128&vers=1460967974696',
  disabled: false,
  country: 'SE',
  locale: 'sv-SE',
  referral_link: 'https://db.tt/NGVu5e5uK1',
  team: {
    id: 'dbtid:AADbFdmDgcSCo5WQ6MesF_q59ltpbjw1lms',
    name: 'Sj\u00f6fartstidningen',
    sharing_policies: {
      shared_folder_member_policy: { '.tag': 'anyone' },
      shared_folder_join_policy: { '.tag': 'from_anyone' },
      shared_link_create_policy: { '.tag': 'default_public' },
    },
    office_addin_policy: { '.tag': 'enabled' },
  },
  team_member_id: 'dbmid:AADhwrnyWESM_u0pImgwyWiB3lHn3ac4ioY',
  is_paired: true,
  account_type: { '.tag': 'business' },
  root_info: {
    '.tag': 'user',
    root_namespace_id: '1194401041',
    home_namespace_id: '1194401041',
  },
};

export { filesListFolder, getAccount, getCurrentAccount };
