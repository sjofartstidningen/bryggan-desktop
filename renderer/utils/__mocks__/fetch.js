import { AbortController } from 'abort-controller';

let __response = {};
const __setResponse = res => (__response = res);

const fetch = (url, { signal }) =>
  new Promise((resolve, reject) => {
    if (signal) {
      signal.addEventListener('abort', () => {
        reject(new Error('Request aborted'));
      });
    }

    setTimeout(() => {
      const response = {
        ok: true,
        status: 200,
        url,
        json: () => Promise.resolve(__response),
      };

      resolve(response);
    });
  });

export { fetch, AbortController, __setResponse };
