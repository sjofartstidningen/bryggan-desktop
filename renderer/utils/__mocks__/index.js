const createSimpleCache = () => {
  const cache = new Map();

  return {
    set: jest.fn((key, value) => cache.set(key, value)),
    get: jest.fn(key => cache.get(key)),
    has: jest.fn(key => cache.has(key)),
    delete: jest.fn(key => cache.delete(key)),
    clear: jest.fn(() => cache.clear()),
  };
};

module.exports = { createSimpleCache };
