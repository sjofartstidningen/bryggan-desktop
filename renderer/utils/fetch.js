const fetch = window && window.fetch ? window.fetch : require('node-fetch');

const AbortController =
  window && window.AbortController
    ? window.AbortController
    : require('abort-controller');

export { fetch, AbortController };
