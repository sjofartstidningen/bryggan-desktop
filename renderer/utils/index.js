import { extname } from 'path';

const sortByType = (a, b) => {
  if (a.type !== b.type) return a.type > b.type ? -1 : 1;

  const aExt = extname(a.name);
  const bExt = extname(b.name);
  if (aExt !== bExt) return aExt > bExt ? 1 : -1;
  if (a.name > b.name) return 1;
  if (a.name < b.name) return -1;
  return 0;
};

export { sortByType };
