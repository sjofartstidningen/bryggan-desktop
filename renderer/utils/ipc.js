import log from 'electron-log';

const callMain = async (...args) => {
  const ipc = require('electron-better-ipc');
  log.verbose(`Call main on channel: ${args[0]}`);
  return ipc.callMain(...args);
};

const answerMain = (...args) => {
  const ipc = require('electron-better-ipc');
  log.verbose(`Setup answer to main on channel: ${args[0]}`);
  return ipc.answerMain(...args);
};

export { callMain, answerMain };
