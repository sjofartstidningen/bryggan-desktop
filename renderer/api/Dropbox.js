import { fetch } from '../utils/fetch';

const normalizeFolderContent = entries =>
  entries
    .map(entry => {
      const type = entry['.tag'];

      switch (type) {
        case 'folder':
          return {
            type,
            id: entry.id,
            name: entry.name,
            path: entry.path_display,
          };
        case 'file':
          return {
            type,
            id: entry.id,
            name: entry.name,
            path: entry.path_display,
            clientModified: entry.client_modified,
            serverModified: entry.server_modified,
            modifiedBy: entry.sharing_info && entry.sharing_info.modified_by,
          };
        default:
          return null;
      }
    })
    .filter(Boolean);

async function listFolder(path, { apiKey, signal } = {}) {
  if (typeof apiKey !== 'string') {
    throw new Error('An api key is required to request folder contents');
  }

  const url = 'https://api.dropboxapi.com/2/files/list_folder';
  const headers = {
    Authorization: `Bearer: ${apiKey}`,
    'Content-Type': 'application/json',
  };

  const body = {
    path,
    recursive: false,
  };

  const response = await fetch(url, {
    method: 'POST',
    headers,
    body: JSON.stringify(body),
    signal,
  });

  if (!response.ok) {
    throw new Error(`Could not fetch folder contents at ${path}`);
  }

  const json = await response.json();

  return { items: normalizeFolderContent(json.entries) };
}

export { listFolder };
