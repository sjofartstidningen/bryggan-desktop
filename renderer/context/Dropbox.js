import React, { createContext, useMemo, useState, useEffect } from 'react';
import * as Dropbox from '../api/Dropbox';

const DropboxContext = createContext();

function Provider({ accessToken: initialAccessToken, children }) {
  const [accessToken, setAccessToken] = useState(() => initialAccessToken);
  const [currentAccount, setCurrentAccount] = useState({});

  const listFolder = (path, { cancelToken, ignoreCache } = {}) => {
    if (!accessToken)
      throw new Error('Api key is required before using the api');
    return Dropbox.listFolder(path, { accessToken, cancelToken, ignoreCache });
  };

  const getAccount = (accountId, { cancelToken, ignoreCache } = {}) => {
    if (!accessToken)
      throw new Error('Api key is required before using the api');
    return Dropbox.getAccount(accountId, {
      accessToken,
      cancelToken,
      ignoreCache,
    });
  };

  const getCurrentAccount = ({ cancelToken } = {}) => {
    if (!accessToken)
      throw new Error('Api key is required before using the api');
    return Dropbox.getCurrentAccount({ accessToken, cancelToken });
  };

  const getToken = async ({ code, clientId, clientSecret }) => {
    const { accessToken } = await Dropbox.getToken({
      code,
      clientId,
      clientSecret,
    });
    setAccessToken(accessToken);
    return { accessToken };
  };

  const revokeToken = async () => {
    if (!accessToken)
      throw new Error('App is not authorized, so no token to revoke');
    await Dropbox.revokeToken({ accessToken });
    setAccessToken(undefined);
  };

  useEffect(
    () => {
      if (accessToken) {
        getCurrentAccount()
          .then(({ account }) => setCurrentAccount(account))
          .catch(() => {});
      }
    },
    [accessToken],
  );

  const contextValue = useMemo(
    () => ({
      accessToken,
      currentAccount,
      listFolder,
      getAccount,
      getCurrentAccount,
      getToken,
      revokeToken,
    }),
    [accessToken, currentAccount],
  );

  return (
    <DropboxContext.Provider value={contextValue}>
      {children}
    </DropboxContext.Provider>
  );
}

export { DropboxContext, Provider };
