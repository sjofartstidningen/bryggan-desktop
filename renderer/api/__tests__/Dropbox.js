import { AbortController, __setResponse } from '../../utils/fetch';
import * as Dropbox from '../Dropbox';
import { filesListFolder } from '../../__fixtures__/Dropbox';

jest.mock('../../utils/fetch.js');
__setResponse(filesListFolder);

describe('Api: Dropbox.listFolder', () => {
  it('should throw if apiKey is not provided', async () => {
    await expect(Dropbox.listFolder('/path')).rejects.toThrow();
  });

  it('should return a normalized array of files and folders of the provided path', async () => {
    const path = '/Tidningen/2018/11';
    const { items } = await Dropbox.listFolder(path, {
      apiKey: 'foo',
    });

    const folder = expect.objectContaining({
      id: expect.any(String),
      type: expect.stringMatching(/folder/gm),
      name: expect.any(String),
      path: expect.stringContaining(path),
    });

    const file = expect.objectContaining({
      id: expect.any(String),
      type: expect.stringMatching(/file/gm),
      name: expect.any(String),
      path: expect.stringContaining(path),
      clientModified: expect.any(String),
      serverModified: expect.any(String),
      modifiedBy: expect.any(String),
    });

    expect(items).toEqual(expect.arrayContaining([folder, file]));
    expect(items).toMatchSnapshot();
  });

  it('should be cancellable', async () => {
    const controller = new AbortController();
    const promise = Dropbox.listFolder('/path', {
      apiKey: 'foo',
      signal: controller.signal,
    });

    controller.abort();

    await expect(promise).rejects.toThrow(/aborted/i);
  });
});
