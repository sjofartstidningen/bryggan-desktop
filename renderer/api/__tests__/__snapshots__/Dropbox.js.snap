// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Api: Dropbox.listFolder should return a normalized array of files and folders of the provided path 1`] = `
Array [
  Object {
    "id": "id:8cfsOlDXQ0AAAAAAAABfZA",
    "name": "A",
    "path": "/Tidningen/2018/11/A",
    "type": "folder",
  },
  Object {
    "id": "id:8cfsOlDXQ0AAAAAAAABfew",
    "name": "D",
    "path": "/Tidningen/2018/11/D",
    "type": "folder",
  },
  Object {
    "id": "id:8cfsOlDXQ0AAAAAAAABf0w",
    "name": "B",
    "path": "/Tidningen/2018/11/B",
    "type": "folder",
  },
  Object {
    "id": "id:8cfsOlDXQ0AAAAAAAABf1A",
    "name": "C",
    "path": "/Tidningen/2018/11/C",
    "type": "folder",
  },
  Object {
    "id": "id:8cfsOlDXQ0AAAAAAAABgdw",
    "name": "Framvagn",
    "path": "/Tidningen/2018/11/Framvagn",
    "type": "folder",
  },
  Object {
    "id": "id:8cfsOlDXQ0AAAAAAAABhzA",
    "name": "till tryckeriet",
    "path": "/Tidningen/2018/11/till tryckeriet",
    "type": "folder",
  },
  Object {
    "clientModified": "2018-11-27T15:14:14Z",
    "id": "id:8cfsOlDXQ0AAAAAAAABf2g",
    "modifiedBy": "dbid:AAAD--pzSalcq73PGbc9Ch9fgh0YPSvL1vQ",
    "name": "ST_18_11.indd",
    "path": "/Tidningen/2018/11/ST_18_11.indd",
    "serverModified": "2018-11-27T15:14:19Z",
    "type": "file",
  },
  Object {
    "clientModified": "2018-11-29T11:54:26Z",
    "id": "id:8cfsOlDXQ0AAAAAAAABgmA",
    "modifiedBy": "dbid:AADI2C0iKz8PvrovDlpZ6DT3KypYuBReQss",
    "name": "Sid 23.pdf",
    "path": "/Tidningen/2018/11/Sid 23.pdf",
    "serverModified": "2018-11-29T11:54:32Z",
    "type": "file",
  },
  Object {
    "clientModified": "2018-12-05T07:53:43Z",
    "id": "id:8cfsOlDXQ0AAAAAAAABgUg",
    "modifiedBy": "dbid:AAAD--pzSalcq73PGbc9Ch9fgh0YPSvL1vQ",
    "name": "ST_11_18_01_Framvagn.indd",
    "path": "/Tidningen/2018/11/ST_11_18_01_Framvagn.indd",
    "serverModified": "2018-12-05T07:53:48Z",
    "type": "file",
  },
  Object {
    "clientModified": "2018-12-05T08:23:00Z",
    "id": "id:8cfsOlDXQ0AAAAAAAABgWA",
    "modifiedBy": "dbid:AAAGsBZeMJWLDSD-bUhgCmyfDrXDDBlV-70",
    "name": "ST_11_18_05_D.indd",
    "path": "/Tidningen/2018/11/ST_11_18_05_D.indd",
    "serverModified": "2018-12-05T08:23:09Z",
    "type": "file",
  },
  Object {
    "clientModified": "2018-12-05T09:03:13Z",
    "id": "id:8cfsOlDXQ0AAAAAAAABgVg",
    "modifiedBy": "dbid:AAAD--pzSalcq73PGbc9Ch9fgh0YPSvL1vQ",
    "name": "ST_11_18_03_B.indd",
    "path": "/Tidningen/2018/11/ST_11_18_03_B.indd",
    "serverModified": "2018-12-05T09:03:18Z",
    "type": "file",
  },
  Object {
    "clientModified": "2018-12-05T09:13:30Z",
    "id": "id:8cfsOlDXQ0AAAAAAAABgVw",
    "modifiedBy": "dbid:AACCoALvX2HMOAqzduK1KNK1IPOb-MKGieI",
    "name": "ST_11_18_04_C.indd",
    "path": "/Tidningen/2018/11/ST_11_18_04_C.indd",
    "serverModified": "2018-12-05T09:13:35Z",
    "type": "file",
  },
  Object {
    "clientModified": "2018-12-05T11:16:46Z",
    "id": "id:8cfsOlDXQ0AAAAAAAABgVA",
    "modifiedBy": "dbid:AAAD--pzSalcq73PGbc9Ch9fgh0YPSvL1vQ",
    "name": "ST_11_18_02_A.indd",
    "path": "/Tidningen/2018/11/ST_11_18_02_A.indd",
    "serverModified": "2018-12-05T11:16:52Z",
    "type": "file",
  },
  Object {
    "clientModified": "2018-12-07T07:16:08Z",
    "id": "id:8cfsOlDXQ0AAAAAAAABiTA",
    "modifiedBy": "dbid:AACrh22Y3qGiQTCRwwE3IHv4GvbjqNoznG8",
    "name": "~st_11_18_01_framva~y0mey8.idlk",
    "path": "/Tidningen/2018/11/~st_11_18_01_framva~y0mey8.idlk",
    "serverModified": "2018-12-07T07:16:14Z",
    "type": "file",
  },
]
`;
